src/components/elements/button/button.styles.ts

```diff
@@ -0,0 +1,107 @@
+'use client';
+
+import { keyframes } from 'styled-components';
+
+import type { ButtonProps } from './button.types';
+
+import { css, styled } from '@/styles';
+
+export const rotate = keyframes`
+    from {
+      transform: rotate(0deg);
+    }
+
+    to {
+      transform: rotate(360deg);
+    }
+`;
+
+export const StyledButton = styled.button<ButtonProps>`
```

Comment:
Take a look at the design, please. It seems the styles are not correct in the implementation. 
_, i.e., The gradient is not in the correct direction._ and this kind of details.